parser/src/ast/mod.rs

```diff
@@ -50,11 +50,55 @@ pub enum SourceSection {
 // TRACE
 // ================================================================================================
 
-/// [TraceCols] contains the main and auxiliary trace columns of the execution trace.
+/// [Trace] contains the main and auxiliary trace columns of the execution trace.
+#[derive(Debug, Eq, PartialEq)]
+pub struct Trace {
+    pub main_cols: Vec<TraceCols>,
+    pub aux_cols: Vec<TraceCols>,
+}
+
+/// [TraceCols] is used to represent a single or a group of columns in the execution trace. For
+/// single columns, the size is 0. For groups, the size is the number of columns in the group.
```

Comment:
Just for my own understanding, is the number of columns sufficient to pinpoint the exact columns (e.g. contiguous and starting from 0)?

Reply:
Regarding this doc comment:
Shouldn't the size be 1 for single columns? (i.e. the number of columns in the group)

Reply:
@grjte Fixed.

@Al-Kindi-0's I'm not sure if you meant this but from your comment it seems like you mean whether column groups need to be contiguous? Currently they are assumed to be contiguous but  I'm wondering if it's possible for some projects that it might be meaningful to have columns in a group that are not contiguous? (@grjte ?). I don't know if any clean syntax is possible to support such use cases though.

Reply:
Thank you @grjte and @tohrnii , the answer of @grjte addresses my question. Regarding whether it might be useful to support non-contiguous cases, I can't think of cases where one might need it at the moment.